### ISSUE TITLE
Coins and CoinsCollector: cyclic reference error in Godot 3.1

### ISSUE THREAD

**Issue by anissen** [GDquest/make-pro-2d-games-with-godot#69]
When trying to run the game on the _very latest from-source_ Godot I get the following error message:

```
Parser Error: Could not fully preload the script, possible cyclic reference or compilation error.
```

`Coins.gd` includes `CoinsCollector.gd` and vice versa.

I expect that the project is made with/for Godot 3.0.x and that this issue is caused by an addition to the source since then.

**Comment by anissen** [comment]
Okay, thanks for the explanation. By the way, I tried resolving the cyclic dependency (by commenting out one of the references) and ran into a _bunch_ of other 3.1-issues.

Should I close the issue (because this repository is 3.0.x-only for now) or keep it open?

**Comment by NathanLovato** [comment]
It's fine for this one. If there's a backwards-compatible way to fix all issues for 3.1, people are welcome to contribute. I just won't do this work now, got enough on my plate already
